src/pages/registration/contributors_tab/components/SelectAffiliationRadioButton.tsx

```diff
@@ -0,0 +1,51 @@
+import CheckCircle from '@mui/icons-material/CheckCircle';
+import CircleOutlined from '@mui/icons-material/CircleOutlined';
+import { IconButton } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { CristinPerson, CristinPersonAffiliation } from '../../../../types/user.types';
+import { dataTestId } from '../../../../utils/dataTestIds';
+
+interface SelectAffiliationRadioButtonProps {
+  personIsSelected: boolean;
+  affiliation: CristinPersonAffiliation;
+  selectedPerson?: CristinPerson;
+  setSelectedPerson: (selectedContributor: CristinPerson | undefined) => void;
+  affiliationIsSelected: boolean;
+}
+
+export const SelectAffiliationRadioButton = ({
+  affiliation,
+  selectedPerson,
+  setSelectedPerson,
+  affiliationIsSelected,
+  personIsSelected,
+}: SelectAffiliationRadioButtonProps) => {
+  const { t } = useTranslation();
+
+  const selectAffiliation = () => {
+    if (!selectedPerson) {
+      return;
+    }
+    const newAffiliations = affiliationIsSelected
+      ? selectedPerson.affiliations.filter((a) => a.organization !== affiliation.organization)
+      : [affiliation];
+
+    const personWithAffiliation: CristinPerson = {
+      ...selectedPerson,
+      affiliations: newAffiliations,
+    };
+    setSelectedPerson(personWithAffiliation);
+  };
+
+  return (
+    <IconButton
```

Comment:
Samme som for checkbox, kan det være verdt å sjekke om å bruke `<Radio>` direkte gir det samme her: https://mui.com/material-ui/react-radio-button/
Bonus ved å bruke Radio og Checkbox er at taksten/label også blir trykkbar, som den egentlig skal være for UU sin del

Reply:
Absolutt, men jeg tror det får bli en senere refaktorering, for nå bruker jeg bare eksisterende kode for å komme raskere i mål.